RUN-89 Add UI profile and search to plugin list endpoint

ProTip opened the pull request:
**Is this a bugfix, or an enhancement? Please describe.**
This enhancement adds UI profile(`iconUrl`, `providerMetadata`) to the public `plugin/list` API endpoint along with the query parameter `service` to filter the results by provider.

**Additional Context**
This is part of ongoing efforts to move functionality out of UI specific endpoints and into the public API.

ProTip commented on rundeckapp/src/test/groovy/rundeck/services/PluginApiServiceSpec.groovy:
```diff
@@ -63,11 +66,14 @@ class PluginApiServiceSpec extends Specification implements ServiceUnitTest<Plug
         def entry = service.providers[0]
 
         then:
+        println(entry)
```
```suggestion
```

gschueler commented on rundeckapp/grails-app/services/rundeck/services/PluginApiService.groovy:
```diff
@@ -268,7 +272,23 @@ class PluginApiService {
-                 enabled      : true]
+                 enabled      : true] as LinkedHashMap<Object, Object>
+
+                if(service != "UI") {
```
need to do an import but preferred to using constant string:

```suggestion
                if(service != ServiceNameConstants.UI) {
```

but, is there a reason to skip UI plugins here?

ProTip replied:
That's a very good question. I brought this bit of logic over from the non-public endpoint that was being used. IIRC I got an error without the check, but in either case I don't enjoy this bit as it's not clear at all why "UI" is an exception. Would need to do further debugging to figure out why.

gschueler commented on rundeckapp/grails-app/controllers/rundeck/controllers/PluginController.groovy:
```diff
@@ -138,10 +138,15 @@ class PluginController extends ControllerBase {
+
         def providers = []
         pluginApiService.listPlugins().each { svc ->
+            if (service && service != svc.service)
```
perhaps this needs an API version bump and version check here

ProTip replied:
This should function the same without the new, optional service parameter supplied. Are we still bumping API version for non-breaking changes?

gschueler replied:
the api bump is so that folks using previous versions don't see the API docs and expect the results returned here. so in this case it's primarily to communicate a required version level for the new features, allowing them to be documented accurately

ProTip commented:
@gschueler I've:

* Added a typed response object using ApiResource
* Implemented a new functional test to ensure the fields are present/missing based on API version endpoint used(and to add a spot for future tests)
* Updated to use service constant

ProTip commented on test/selenium/src/__tests__/api/plugins.test.ts:
```diff
@@ -0,0 +1,45 @@
+        expect(found).toBeFalsy()
+    })
+
+    it('Returns v40 fieldd', async () => {
```
```suggestion
    it('Returns v40 field', async () => {
```
